js/coinsbit.js

```diff
@@ -0,0 +1,682 @@
+const { TICK_SIZE } = require ('./base/functions/number');
+const Precise = require ('./base/Precise');
+// function c(o){console.log(o);}
+// ---------------------------------------------------------------------------
+
+module.exports = class coinsbit extends Exchange {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'coinsbit',
+            'name': 'Coinsbit',
+            'countries': [ 'EE' ], // Estonia
+            // 'rateLimit': 50, // default rate limit is 20 times per second
+            'version': 1,
+            'certified': false,
+            'has': {
+                'addMargin': undefined,
+                'cancelAllOrders': undefined,
+                'cancelOrder': undefined,
+                'createMarketOrder': undefined,
+                'createOrder': undefined,
+                'fetchBalance': undefined,
+                'fetchCanceledOrders': undefined,
+                'fetchClosedOrders': undefined,
+                // 'fetchCurrencies' doesn't exist, and will be 'emulated'
+                'fetchDepositAddress': undefined,
+                'fetchDeposits': undefined,
+                'fetchFundingRateHistory': undefined,
+                'fetchMarkets': true,
+                // 'fetchMarketsByType': <-- i think fetchMarketsByType makes things complex, and might be totally removed from ccxt. There should be one 'fetchMarkets' imho. you can see another comment related to this matter under 'checkExchangeType' function down below.
+                'fetchMyTrades': undefined,
+                'fetchOHLCV': true,
+                'fetchOpenOrders': undefined,
+                'fetchOrder': undefined,
+                'fetchOrderBook': true,
+                'fetchOrderTrades': undefined,
+                'fetchPosition': undefined,
+                'fetchPositions': undefined,
+                'fetchStatus': 'emulated',
+                'fetchTicker': true,
+                'fetchTickers': true,
+                'fetchTime': 'emulated',
+                'fetchTrades': undefined,
+                'fetchWithdrawals': undefined,
+                'reduceMargin': undefined,
+                'setLeverage': undefined,
+                'withdraw': undefined,
+            },
+            'timeframes': {
+                // This exchange has only 4 intervals of second-scale.
+                //    ________________________________________
+                //    | interval | works for diapason up to  |
+                //    |--------------------------------------|
+                //    | 60       | 12 hours (43200 seconds)  |
+                //    | 1800     | 24 hours (86400 seconds)  |
+                //    | 3600     | 7 days (7*86400 seconds)  |
+                //    | 86400    | any interval              |
+                //    |__________|___________________________|
+                '1m': 60,
```

Comment:
I would remove that comment, cause it's obvious from the values below.

Reply:
Well, there is a bit problem. The comment block I've included (the visual side doesnt matter here, but it's meaning) is that comment has to be moved either inside fetchOHLCV, as that comment/information is nowhere available (neither in docs) and is for informative purposes. If that comment was not anywhere, then user doesn't know how much period he can request with i.e. 1 min bars (that comment says that maximum is 12 hours). Otherwise, if we remove that part, from where it will be obvious that 1min bars work at max for 12 hours diapason?

Reply:
CCXT does not unify the maximum OHLCV length yet.
I'd simply suggest to rewrite it like this, if you want to keep that information in comments:

```
            'timeframes': {
                '1m': 60, // max 12 hours into the past
                '30m': 1800, // max 24 hours
                '1h': 3600, // max 7 days
                '1d': 86400, // any range
            },

Reply:
k